arlas-persistence-core/src/main/java/io/arlas/persistence/server/impl/GoogleFirestorePersistenceServiceImpl.java

```diff
@@ -373,6 +373,42 @@ private Optional<Data> getByZoneKeyOrga(String zone, String key, List<String> or
         }
     }
 
+    private Optional<Data> getByZoneKeyOrga(String zone, String key, IdentityParam idp) throws ArlasException {
+
+        try {
+             // get the data matching zone+key whatever the organisation
+            List<Data> res = db.collection(this.collection)
+                    .whereEqualTo(Data.zoneColumn, zone)
+                    .whereEqualTo(Data.keyColumn, key)
+                    .get().get()
+                    .getDocuments()
+                    .stream()
+                    .map(d -> {
+                        try {
+                            return toData(d.getId(), d);
+                        } catch (NotFoundException e) { //can't happen in this case
+                            return null;
+                        }
+                    })
+                    .filter(Objects::nonNull)
+                    // if the data's organisation is the org of the user
+                    .filter(d -> idp.organisation.contains(d.getDocOrganization())
```

Comment:
in this case, we might endup fetching more zones in the case
of an anonymous idp no ?